--- sphinxcontrib-issuetracker-0.10.1/doc/conf.py.orig	2012-06-18 18:33:56.000000000 +0200
+++ sphinxcontrib-issuetracker-0.10.1/doc/conf.py	2012-08-31 21:28:36.154850626 +0200
@@ -54,8 +54,8 @@
 html_theme = 'default'
 html_static_path = []
 
-intersphinx_mapping = {'python': ('http://docs.python.org/', None),
-                       'sphinx': ('http://sphinx.pocoo.org/', None)}
+intersphinx_mapping = {'python': ('http://docs.python.org/', 'python-objects.inv'),
+                       'sphinx': ('http://sphinx.pocoo.org/', 'sphinx-objects.inv')}
 
 issuetracker = 'github'
 issuetracker_project = 'lunaryorn/sphinxcontrib-issuetracker'
--- sphinxcontrib-issuetracker-0.10.1/doc/conf.py.orig	2012-09-01 06:58:57.590804016 +0200
+++ sphinxcontrib-issuetracker-0.10.1/doc/conf.py	2012-09-01 07:49:41.897407226 +0200
@@ -59,6 +59,7 @@
 
 issuetracker = 'github'
 issuetracker_project = 'lunaryorn/sphinxcontrib-issuetracker'
+issuetracker_plaintext_issues = False
 
 EVENT_SIG_RE = re.compile(r'([a-zA-Z-]+)\s*\((.*)\)')
 
