--- sphinxcontrib-issuetracker-0.11/tests/conftest.py.orig	2012-06-28 17:12:27.000000000 +0200
+++ sphinxcontrib-issuetracker-0.11/tests/conftest.py	2023-11-22 19:33:45.157861467 +0100
@@ -31,7 +31,7 @@ from mock import Mock
 from docutils import nodes
 from sphinx.application import Sphinx
 from sphinx.builders.html import StandaloneHTMLBuilder
-from sphinx.environment import SphinxStandaloneReader
+from sphinx.io import SphinxStandaloneReader
 from sphinx.addnodes import pending_xref
 
 from sphinxcontrib.issuetracker import Issue, IssueReferences
--- sphinxcontrib-issuetracker-0.11/doc/conf.py.orig	2023-12-04 08:00:14.880894958 +0100
+++ sphinxcontrib-issuetracker-0.11/doc/conf.py	2023-12-04 08:12:05.450378806 +0100
@@ -74,6 +74,6 @@ def parse_event(env, sig, signode):
 
 
 def setup(app):
-    app.add_description_unit('confval', 'confval',
+    app.add_object_type('confval', 'confval',
                              'pair: %s; configuration value')
-    app.add_description_unit('event', 'event', 'pair: %s; event', parse_event)
+    app.add_object_type('event', 'event', 'pair: %s; event', parse_event)
--- sphinxcontrib-issuetracker-0.11/sphinxcontrib/issuetracker/__init__.py.orig	2012-12-26 00:52:11.000000000 +0100
+++ sphinxcontrib-issuetracker-0.11/sphinxcontrib/issuetracker/__init__.py	2023-12-04 20:53:34.626192634 +0100
@@ -301,12 +301,10 @@ def init_transformer(app):
 def copy_stylesheet(app, exception):
     if app.builder.name != 'html' or exception:
         return
-    app.info(bold('Copying issuetracker stylesheet... '), nonl=True)
     dest = path.join(app.builder.outdir, '_static', 'issuetracker.css')
     source = path.join(path.abspath(path.dirname(__file__)),
                        'issuetracker.css')
     copyfile(source, dest)
-    app.info('done')
 
 
 def setup(app):
